### PR TITLE
Qualify module names by paths

### PR DESCRIPTION
**Pull request by skingo**
Modules are currently just the names of the files.
This leads to long search times and possible confusion between files of the same name.
To avoid this, we should qualify the names of all modules by their paths from the root (the given "library path"), e.g. `Function` becomes `Codata.Function`.

### CODE REVIEW

File: std/Codata/CoMaybe.duo [hunk 1]
```diff
@@ -0,0 +1,79 @@
+module Codata.CoMaybe;
+
+import Codata.Function;
+import Data.Cofunction;
+import Data.NegationP;
+import Data.Bool;
+import Prim.I64;
+
+-- | The Maybe type encapsulates optional values.
```
Comment (BinderDavid) [inline std/Codata/CoMaybe.duo:9]: We probably shouldn't duplicate Data.Maybe here.

File: src/Syntax/CST/Program.hs [hunk 2]
```diff
@@ -291,10 +298,31 @@ instance Show Declaration where
...
-    -- ^ The absolute filepath of the module.
+  , mod_libpath :: FilePath
+    -- ^ The absolute filepath of the library of the module.
   , mod_decls :: [Declaration]
     -- ^ The declarations contained in the module.
   }
 
 deriving instance Show Module
+
+adjustModulePath :: Module -> FilePath -> Either (NE.NonEmpty Error) Module
```
Comment (BinderDavid) [inline src/Syntax/CST/Program.hs:309]: Kannst du hier ein Kommentar vor die Funktion packen mit Beschreibung was die Funktion macht, evtl mit einem Beispiel?

File: std/Codata/CoMaybe.duo (hunk 1 above)
Comment (skingo) [reply std/Codata/CoMaybe.duo:9]: ah, hab ich das aus Versehen commitet?
Kann raus.

File: src/Syntax/CST/Program.hs (hunk 2 above)
Comment (skingo) [reply src/Syntax/CST/Program.hs:313]: so?

**Review by BinderDavid — APPROVED** [review]

**Pull request merged**